File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int lb=0;
        int rb=height.size()-1;
        int maxwater=0;
        while(lb<rb){
            int wd=rb-lb;
            int ht=min(height[lb],height[rb]);
            int curwater=wd*ht;
            maxwater=max(maxwater,curwater);
            height[lb]<height[rb]?lb++:rb--;
        }
        return maxwater;
        
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] h) {
    int a=1;
    int max=0;
    int i=0;
    int j=h.length-1;
    while(i<j){
    int min=Math.min(h[i],h[j]);
    a=min*(j-i);
    if(a>max)
    {
max=a;}
if(h[i]<h[j]){
i++;
}else{
j--;
}
}
return max;
        
    }
}